### PR TITLE
Add an explicit base class pointer to avoid potential breakage.

### LINKED ISSUE #5499: Error in Building route_guide example application
Dear All

Ubuntu Machine Details:

Linux developer-VirtualBox 3.8.0-35-generic #52~precise1-Ubuntu SMP Thu Jan 30 17:27:28 UTC 2014 i686 i686 i386 GNU/Linux
root@developer-VirtualBox:/home/developer# g++ --version
g++ (Ubuntu/Linaro 4.6.3-1ubuntu5) 4.6.3
Copyright (C) 2011 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

root@developer-VirtualBox:/home/developer# g++ --version
g++ (Ubuntu/Linaro 4.6.3-1ubuntu5) 4.6.3
Copyright (C) 2011 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Compiled and installed protobuf-cpp-3.0.0-beta-2 and grpc-master packages in Ubuntu.
Generated GRPC and Protobuf source for route_guide example in Ubuntu Machine.

QNX QDE:
Compiled both protobuf and grpc unsecure version.
While compiling the sample application getting error
**call.h:300:5: error: use of deleted function 'void std::unique_ptr<_Ty, _Dx>::reset(_Ptr2) [with _Ptr2 =**

17:10:04 **** Incremental Build of configuration Debug for project routeguideClient ****
Info: Internal Builder is used for build
qcc -o helper.o "..\\helper.cc" -V4.7.3,gcc_ntox86_cpp -w1 "-ID:\\Ramesh\\\POC_WS\\OS_ProtoBuf_x86\\src" "-ID:\\Ramesh\\\POC_WS\\OS_GRPC_CPP_x86\\include" -c -O0 -g -std=c++0x 
qcc -o route_guide.grpc.pb.o "..\\route_guide.grpc.pb.cc" -V4.7.3,gcc_ntox86_cpp -w1 "-ID:\\Ramesh\\\POC_WS\\OS_ProtoBuf_x86\\src" "-ID:\\Ramesh\\\POC_WS\\OS_GRPC_CPP_x86\\include" -c -O0 -g -std=c++0x 
In file included from D:\Ramesh\POC_WS\OS_ProtoBuf_x86\src/google/protobuf/descriptor.h:57:0,
                 from D:\Ramesh\POC_WS\OS_ProtoBuf_x86\src/google/protobuf/message.h:122,
                 from ..\route_guide.pb.h:26,
                 from ..\route_guide.grpc.pb.cc:5:
D:\Ramesh\POC_WS\OS_GRPC_CPP_x86\include/grpc++/impl/codegen/call.h: In instantiation of 'void grpc::CallOpGenericRecvMessage::RecvMessage(R*) [with R = routeguide::RouteSummary]':
D:\Ramesh\POC_WS\OS_GRPC_CPP_x86\include/grpc++/impl/codegen/sync_stream.h:188:5:   required from 'grpc::ClientWriter<W>::ClientWriter(grpc::ChannelInterface*, const grpc::RpcMethod&, grpc::ClientContext*, R*) [with R = routeguide::RouteSummary; W = routeguide::Point]'
..\route_guide.grpc.pb.cc:54:114:   required from here
C:/qnx660/target/qnx6/usr/include/cpp/memory:1726:8: error: 'void std::unique_ptr<_Ty, _Dx>::reset(_Ptr2) [with _Ptr2 = grpc::CallOpGenericRecvMessageHelper::DeserializeFuncType<routeguide::RouteSummary>*; _Ty = grpc::CallOpGenericRecvMessageHelper::DeserializeFunc; _Dx = std::default_delete<grpc::CallOpGenericRecvMessageHelper::DeserializeFunc>]' is private
In file included from D:\Ramesh\POC_WS\OS_GRPC_CPP_x86\include/grpc++/impl/codegen/async_stream.h:38:0,
                 from ..\route_guide.grpc.pb.h:9,
                 from ..\route_guide.grpc.pb.cc:6:
D:\Ramesh\POC_WS\OS_GRPC_CPP_x86\include/grpc++/impl/codegen/call.h:300:5: error: within this context
D:\Ramesh\POC_WS\OS_GRPC_CPP_x86\include/grpc++/impl/codegen/call.h:300:5: error: use of deleted function 'void std::unique_ptr<_Ty, _Dx>::reset(_Ptr2) [with _Ptr2 = grpc::CallOpGenericRecvMessageHelper::DeserializeFuncType<routeguide::RouteSummary>*; _Ty = grpc::CallOpGenericRecvMessageHelper::DeserializeFunc; _Dx = std::default_delete<grpc::CallOpGenericRecvMessageHelper::DeserializeFunc>]'
In file included from D:\Ramesh\POC_WS\OS_ProtoBuf_x86\src/google/protobuf/descriptor.h:57:0,
                 from D:\Ramesh\POC_WS\OS_ProtoBuf_x86\src/google/protobuf/message.h:122,
                 from ..\route_guide.pb.h:26,
                 from ..\route_guide.grpc.pb.cc:5:
C:/qnx660/target/qnx6/usr/include/cpp/memory:1726:8: error: declared here
cc: C:/qnx660/host/win32/x86/usr/lib/gcc/i486-pc-nto-qnx6.6.0/4.7.3/cc1plus caught signal 1

### PR DESCRIPTION
**Pull request by yang-g**
Should fix #5499.

### CODE REVIEW

File: include/grpc++/impl/codegen/call.h [hunk 1]
```diff
@@ -329,8 +329,9 @@ class CallOpGenericRecvMessage {
 
   template <class R>
   void RecvMessage(R* message) {
-    deserialize_.reset(
-        new CallOpGenericRecvMessageHelper::DeserializeFuncType<R>(message));
+    CallOpGenericRecvMessageHelper::DeserializeFunc* func =
```
Comment (dklempner) [inline include/grpc++/impl/codegen/call.h]: You should add a comment as to why this is necessary.

Comment (yang-g) [inline include/grpc++/impl/codegen/call.h]: Done.

**Comment by yang-g** [comment]
Green enough?

**Comment by yang-g** [comment]
Ping.

**Comment by dklempner** [comment]
LGTM

**Pull request merged**